### ISSUE TITLE
Review django integration with AngularJS

### ISSUE THREAD

**Issue by pfitzpaddy** [iMMAP-Afg/ngmDevWorkflow#18]
Will need to determine if django-angularJS (as per the [thinkster tutorial](https://thinkster.io/django-angularjs-tutorial)) is a good approach or straight django (viewflow) for our application framework. In particular whats of interest is if a viewflow django model form can be successfully integrated in an angular app without the need for additional js fluff or coding.

**Comment by pfitzpaddy** [comment]
One example is [django-angular](https://github.com/jrief/django-angular), with associated doco [here](http://django-angular.readthedocs.io/en/latest/) and demo [here](http://django-angular.awesto.com/classic_form/). Probelm is;
* Uses bootstrap (not a big deal, can be converted)
* Looks like creating a form requires additional coding on top of the django-model form, which is what we are trying to avoid. We want the form to be automatically generated from a config and managed by django as much as possible.

In other words;
* Run demos of [danjo-angular](http://django-angular.readthedocs.io/en/latest/demos.html)
* Run [viewflow](http://docs.viewflow.io/index.html)

Determine which form requires less maintenance. This might in turn determine our framework. At the moment i was hoping to implement angular widgets, might end up being a total django framework.

**Comment by pfitzpaddy** [comment]
Adding a link to a tutorial i will also look at [here](http://glynjackson.org/weblog/tutorial-using-angularjs-django/), but most likely django-angular docs should cover it

**Comment by pfitzpaddy** [comment]
[awesome-django](http://awesome-django.com/) thanks @drfaustusfade

**Comment by pfitzpaddy** [comment]
Will go with a straight django implementation as the resulting code will be modular and easier to manage. Nice tutorial to get the ball rolling is [here](https://tutorial.djangogirls.org/en/). One plus point is the amount of django libraries that we can plug-in, such as user profiles/user management [here](http://django-userena.readthedocs.io/en/latest/). We will be able to re-use as many libraries as possible to keep our codebase down, also documentation, test cases etc will be easier to implement. The attached PDF is old but points out the philosophy well.
[reusable_apps.pdf](https://github.com/iMMAP-Afg/ngmDevWorkflow/files/1104371/reusable_apps.pdf)
